### PR TITLE
PP-584 (REL-005): Add secure transfer in the Collector

### PR DESCRIPTION
**Pull request by AndresQuijano**
## What

- Change the use of transfer from ERC20 to .call.

## Why

- Security bug found in REL-005.

## Refs

- [PP-584](https://rsklabs.atlassian.net/browse/PP-584)

### CODE REVIEW

**Review by antomor — CHANGES_REQUESTED** [review]
@AndresQuijano the changes look good to me and we should add also some successful scenarios in the tests, to check that the transaction doesn't fail if the transfer is successful.

Furthermore, for the sake of curiosity, have we analyzed the gas consumption change of this addition? I'm wondering if not using the openzeppelin contracts but doing the checks directly would be cheaper.

**Comment by AndresQuijano** [comment]
> @AndresQuijano the changes look good to me and we should add also some successful scenarios in the tests, to check that the transaction doesn't fail if the transfer is successful.
> 
> Furthermore, for the sake of curiosity, have we analyzed the gas consumption change of this addition? I'm wondering if not using the openzeppelin contracts but doing the checks directly would be cheaper.

As we talked before, I think the success cases are covered by previous test.
Addiotionally this is the report of gas consumption using the different alternatives.
image

**Review by ulasanil — APPROVED** [review]
LGTM

**Review by jurajpiar — APPROVED** [review]
Just a minor improvement.

File: test/collector.test.ts [hunk 1]
```diff
@@ -285,6 +287,41 @@ describe('Collector', () => {
...
+                partners,
+                oldRemainder.address
+            ]);
+
+            await expect(
+                collector.updateRemainderAddress(newRemainder.address, {
+                    from: owner.address,
+                    gasLimit: 100000
+                })
+            ).to.be.reverted;
```
[the remark this answers is not in the document]
Comment (AndresQuijano) [reply test/collector.test.ts]: I agree. Done.

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**